tests/unit/dialog/core.js

```diff
@@ -70,11 +70,12 @@ QUnit.test( "title id", function( assert ) {
 } );
 
 QUnit.test( "ARIA", function( assert ) {
-	assert.expect( 4 );
+	assert.expect( 5 );
 
 	var element = $( "<div>" ).dialog(),
 		wrapper = element.dialog( "widget" );
 	assert.equal( wrapper.attr( "role" ), "dialog", "dialog role" );
+	assert.equal( wrapper.attr( "aria-modal" ), "true", "aria-modal attribute" );
```

Comment:
Discussed with @rpkoller, this is a possible change to the tests given the  suggested changes in https://github.com/jquery/jquery-ui/pull/2257/files#r1619122875.

``` javascript
if (element.dialog( "option", "modal" )) {
  assert.equal( wrapper.attr( "aria-modal" ), "true", "aria-modal attribute" );
} else {
  assert.ok( !element.attr( "aria-modal" ), "aria-modal not set" );
}
```

Reply:
`if`s like that are not a good practice in test - you'd still be testing only one of them and the other would be left untested. We need both tests to run. We need one dialog with the `modal` option set - and then verify the `aria-modal` attribute is set to `"true"`, and another one without the `modal` option that verifies the attribute has not been set.

This test is failing right now because the dialog was created without the `modal` option.

I'd recommend creating a separate test just after this one titled `ARIA, aria-modal` where you create three dialogs - one with the `modal` option set to `true`, one to `false` and one with no such option passed, and verify proper `aria-model` value or presence for each of them.

This `"ARIA"` test should be reverted to its initial form, all new assertions would be added to this new test.